aspnetcore/fundamentals/logging.md

```diff
@@ -19,70 +19,95 @@ By [Steve Smith](http://ardalis.com) and [Tom Dykstra](https://github.com/tdykst
 
 ASP.NET Core [dependency injection](dependency-injection.md) (DI) provides the `ILoggerFactory` instance. The `AddConsole` and `AddDebug` extension methods are defined in the [Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console/) and [Microsoft.Extensions.Logging.Debug](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Debug/) packages. Each extension method calls the `ILoggerFactory.AddProvider` method, passing in an instance of the provider. 
 
 > [!NOTE]
 > The sample application for this article adds logging providers in the `Configure` method of the `Startup` class. If you want to get log output from code that executes earlier, add logging providers in the `Startup` class constructor instead. 
 
-You'll find information about each [built-in logging provider](#built-in-logging-providers) and links to [third-party logging providers](#third-party-logging-providers) later in the article.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-## How to create logs
+A logging provider takes the messages that you create with an `ILogger` object and displays or stores them. For example, the Console provider displays messages on the console, and the Azure App Service provider can store them in Azure blob storage.
+
+To use a provider, call the provider's `Add<ProviderName>` extension method in *Program.cs*:
 
-To create logs, get an `ILogger` object from DI and store it in a field, then call logging methods on that logger object.
+[!code-csharp[](logging/sample2/Program.cs?name=snippet_ExpandDefault&highlight=16,17)]
 
-[!code-csharp[](logging/sample/src/TodoApi/Controllers/TodoController.cs?name=snippet_LoggerDI&highlight=4,7,10)]
+The default project template sets up logging the way you see it in the preceding code, but the `ConfigureLogging` call is done by the `CreateDefaultBuilder` method. Here's the code in *Program.cs* that is created by project templates:
 
-[!code-csharp[](logging/sample/src/TodoApi/Controllers/TodoController.cs?name=snippet_CallLogMethods&highlight=3,7)]
+[!code-csharp[](logging/sample2/Program.cs?name=snippet_TemplateCode&highlight=7)]
 
-This example requests `ILogger<TodoController>` from DI to specify the `TodoController` class as the *category* of logs that are created with the logger.  Categories are explained [later in this article](#log-category).
+---
+
+You'll find information about each [built-in logging provider](#built-in-logging-providers) and links to [third-party logging providers](#third-party-logging-providers) later in the article.
 
 ## Sample logging output
 
-With the sample code shown above, you'll see logs in the console when you run from the command line, and in the Debug window when you run in Visual Studio in Debug mode. 
+With the sample code shown in the preceding section, you'll see logs in the console when you run from the command line, and in the Debug window when you run in Visual Studio in Debug mode. 
```

Comment:
This sentence seems a bit long. Maybe something like the following would be better:

With the sample code shown in the preceding section, you'll see:
- Logs in the console when you run from the command line
- Logs in the Debug window when you run in Visual Studio in Debug mode